packages/react-dev-utils/WebpackDevServerUtils.js

```diff
@@ -190,16 +191,28 @@ function createCompiler({
 
       const messages = await tsMessagesPromise;
       clearTimeout(delayedMsg);
-      statsData.errors.push(...messages.errors);
+      if (tscCompileOnError) {
```

Comment:
It seems like the value of this flag is in development only, as pushing warnings instead of errors will still cause a build failure of a production build.

I think we should only respect this flag in development mode, and can document as such in `advanced-configuration.md`. @mrmckeb thoughts?